src/main/java/com/aws/greengrass/logging/impl/LogManager.java

```diff
@@ -118,82 +119,96 @@ public static void setRoot(Path newPath) {
     }
 
     /**
-     * Sets effective configuration.
+     * Reset the non-null fields of given config for all loggers back to the default value.
+     * Supported RESET topics:
+     *      level, format, outputType, fileSizeKB, totalLogsSizeKB, outputDirectory
      *
-     * @param loggerConfiguration configuration set in yaml.
+     * @param resetTopicName name of the topic to reset
      */
-    public static void setEffectiveConfig(LoggerConfiguration loggerConfiguration) {
-        LogConfig rootConfig = LogConfig.getInstance();
-        if (loggerConfiguration.getFileName() == null || loggerConfiguration.getFileName().trim().isEmpty()) {
-            loggerConfiguration.setFileName(rootConfig.getFileName());
-        }
-        if (loggerConfiguration.getFileSizeKB() == -1) {
-            loggerConfiguration.setFileSizeKB(rootConfig.getFileSizeKB());
-        }
-        if (loggerConfiguration.getTotalLogsSizeKB() == -1) {
-            loggerConfiguration.setTotalLogsSizeKB(rootConfig.getTotalLogStoreSizeKB());
-        }
-        if (loggerConfiguration.getFormat() == null) {
-            loggerConfiguration.setFormat(rootConfig.getFormat());
-        }
-        if (loggerConfiguration.getLevel() == null) {
-            loggerConfiguration.setLevel(rootConfig.getLevel());
-        }
-        if (loggerConfiguration.getOutputType() == null) {
-            loggerConfiguration.setOutputType(rootConfig.getStore());
-        }
-        if (loggerConfiguration.getOutputDirectory() == null && LogStore.FILE
-                .equals(loggerConfiguration.getOutputType())) {
-            loggerConfiguration.setOutputDirectory(rootConfig.getStoreDirectory().toString());
+    public static void resetAllLoggers(String resetTopicName) {
+        // Constructing a new PersistenceConfig here because its constructor can check system properties for logging
+        // configs, which should be used as default
+        PersistenceConfig defaultConfig = new PersistenceConfig(LOG_FILE_EXTENSION, LOGS_DIRECTORY);
+        if (resetTopicName == null) {  // reset all to default
+            reconfigureAllLoggers(new LogConfigUpdate(defaultConfig));
+        } else {  // reset individual config
+            LogConfigUpdate configToApply = LogConfigUpdate.builder().build();
+            switch (resetTopicName) {
+                case "level":
+                    configToApply.setLevel(defaultConfig.getLevel());
```

Comment:
nit - it's weird that we have `builder` and `setter` together. One defeats the purpose of the other one. Please double check if we could remove setter completely.

Reply:
Hmm good point. If we decide to use builder, we shouldn't have setters to ensure immutability. I can change it here but our code base already has many classes with both `@Builder` and setter (provided by `@Data`) (e.g. ComponentArtifact, DeploymentInformation, and many more...). Those can be unintentional because we were not aware `@Data` includes setters, or maybe we did it for convenience...

Reply:
I guess for many cases, making data class immutable is too cumbersome to justify the immutability benefit it provides...